gustaf/spline/create.py

```diff
@@ -254,6 +255,397 @@ def revolved(
+                    [1., 0.], [-0.5, 0.5], [0., 1.], [0.5, 0.5]
+            ]
+    ) * radius
+    weights = np.tile([1., 1 / np.sqrt(2)], 5)[:-1]
+
+    return RationalBezier(
+            degrees=degrees, control_points=control_points, weights=weights
+    )
+
+
+def disk(outer_radius, inner_radius=None, angle=360., n_knot_spans=4):
+    """Surface spline describing a potentially hollow disk with quadratic
+    degree along curved dimension and linear along thickness.
+    The angle describes the returned part of the disk.
+
+    Parameters
+    ----------
+    outer_radius : float
+      Outer radius of the disk
+    inner_radius : float, optional
+      Inner radius of the disk, in case of hollow disk, by default 0.
+    angle : float, optional
+      Rotational angle, by default 360. describing a complete revolution
+    n_knot_spans : int, optional
+      Number of knot spans, by default 4
+
+    Returns
+    -------
+    gustaf.NURBS
+      Surface NURBS of degrees (1,2)
+    """
+
+    from gustaf.spline import NURBS
+
+    if inner_radius is None:
+        inner_radius = 0.
+
+    cps = np.array([[inner_radius, 0.], [outer_radius, 0.]])
+    weights = np.ones([cps.shape[0]])
+    knots = np.repeat([0., 1.], 2)
+
+    return NURBS(
+            control_points=cps,
+            knot_vectors=[knots],
+            degrees=[1],
+            weights=weights
+    ).create.revolved(angle=angle, n_knot_spans=n_knot_spans)
+
+
+def torus(
+        torus_radius,
+        section_outer_radius,
+        section_inner_radius=None,
+        torus_angle=None,
+        section_angle=None,
+        section_n_knot_spans=4,
+        torus_n_knot_spans=4
+):
+    """Creates a volumetric NURBS spline describing a torus revolved around
+    the x-axis.
+    Possible cross-sections are plate, disk (yielding a tube) and section of
+    a disk.
+
+    Parameters
+    ----------
+    torus_radius : float
+      Radius of the torus
+    section_outer_radius : float
+      Radius of the section of the torus
+    section_inner_radius : float, optional
+      Inner radius in case of hollow torus, by default 0.
+    torus_angle : float, optional
+      Rotational anlge of the torus, by default None, giving a complete
+      revolution
+    section_angle : float, optional
+      Rotational angle, by default None, yielding a complete revolution
+    section_n_knot_spans : float, optional
+      Number of knot spans along the cross-section, by default 4
+    torus_n_knot_spans : float, optional
+      Number of knot spans along the torus, by default 4
+
+    Returns
+    -------
+    gustaf.NURBS
+      Volumetric spline in the shape of a torus with degrees (1,2,2)
+    """
+    if torus_angle is None:
+        torus_angle = 360
+    if section_angle is None:
+        section_angle = 360
+    if section_inner_radius is None:
+        section_inner_radius = 0
+
+    # Create the cross-section
+    if section_angle == 360 and section_inner_radius == 0:
```

Comment:
Float Comparisons are always kind of dangerous, maybe put an additional boolean if angle is None